base/db/R/insert.format.vars.R

```diff
@@ -1,28 +1,67 @@
 #' Insert Format and Format-Variable Records 
-#' @param con Bety connection object
+#'
+#' @param con SQL connection to BETYdb
+#' @param format_notes Character string that describes format.
+#' @param format_name Character string 
+#' @param header Boolean that indicates the presence of a header in the format 
+#' @param skip Integer that indicates the number of lines to skip in the header. Defaults to "" 
+#' @param mimetype_id Integer 
 #' @param formats_variables_df A 'data.frame' consisting of entries that correspond to columns in the formats-variables table
-#' @param formats_df A 'data.frame' consisting of entries that correspond to columns in the formats table
+#' @details It is very important that the formats_variables_df 'data.frame' be structured in a specific format so that the SQL query functions properly. First, all arguments should be passed as vectors so that each entry will correspond with a specific row.
+#' Variables: 
+#' \enumerate{
+#' \item variable_id: vector of integers
+#' \item name: vector of character strings
+#' \item unit: vector of character strings
+#' \item storage_type: vector of character strings
+#' \item column_number: vector of integers
+#' }
 #' @author Liam Burke (liam.burke24@gmail.com)
-#' @return Data frame: Inner join of SQL table and input data frame (as unevaluated "lazy query" table) 
+#' @return format_id
 #' @export
 #' @examples
 #' \dontrun{
 #' bety <- betyConnect
-#'   insert.format.vars(con = bety$con, formats_df, formats_variables_df)
+#' 
+#' formats_variables_df <- tibble::tibble(
+#'        variable_id = c(327, 2.98e+02), # integer 
+#'        name = c("New Name", "Other New Name"), # 
+#'        unit = c("Yoonits", "Other Units"),
+#'        storage_type = c("%j", "%Y"),
+#'        column_number = c(as.integer(22), as.integer(24)),
+#'  )
+#'   insert.format.vars(con = bety$con, format_notes = "Info about format", format_name = "New Format", header = TRUE, skip = 2, mimetype_id = 1090, formats_variables_df)
 #' }
-insert.format.vars <- function(con, formats_df, formats_variables_df = NULL){
-  if(is.null(formats_variables_df)){
-    ## Only insert format record
+insert.format.vars <- function(con, format_notes, format_name, header, skip = "", mimetype_id, formats_variables_df = NULL){
+  
+    formats_df <- tibble::tibble(
+      header = as.character(header),
+      skip = skip,
+      mimetype_id = mimetype_id,
+      notes = format_notes,
+      name = format_name,
+      stringsAsFactors = FALSE
+    )
```

Comment:
As noted previously, you might want to perform some basic checks on these values before inserting them. For example, if skip is present it needs to be numeric, header needs to be boolean, notes and names need to be character. Mimetype_id needs to be numeric, and if you want to be really good you'll make sure that ID exists before you insert the record. Similarly, you might want to check the format name doesn't already exist.  In general, inserting junk into the db creates problems that are remarkably time consuming to clean up, so it's worth the extra effort on inserts to make sure that what you're inserting makes sense.

Reply:
I perform basic checks on all format types of all inputs and check if the format name exists in the new commits. Can you please clarify what you mean by making sure that the ID exists before inserting the record? Originally I thought you meant the mimetype_id, but this should exist already. I would appreciate some clarification. Thanks!

Reply:
Got it. That was kind of counter intuitive since I keep forgetting that I'm building this function for other applications besides the shiny app. Testing if the variable ID exists is pointless in my app since all the ID's are resolved by the user selecting a variable name.

Reply:
It actually doesn't make sense in this application since I am not passing the variable name. The user selects BETY variables from a dropdown menu and then the app resolves the variable_id. I included a test but I placed it in a conditional `suppress` statement so that I can suppress it in my app.